File: app/src/main/java/com/example/bettersleep/entity/SleepDao.java
package com.example.bettersleep.entity;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface SleepDao {
    @Query("SELECT * FROM sleep_table ORDER BY id ASC")
    LiveData<List<Sleep>> getOrderAscSleep();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(Sleep sleep);

    @Query("DELETE FROM sleep_table")
    void deleteAll();
}

File: app/src/main/java/com/nursyah/bettersleep/entity/SleepDao.kt
package com.nursyah.bettersleep.entity

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface SleepDao {
    @get:Query("SELECT * FROM sleep_table ORDER BY id DESC")
    val allData: LiveData<List<Sleep>>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(sleep: Sleep?)

    @Query("DELETE FROM sleep_table")
    fun deleteAll()
}